tests/test_templates.py

```diff
@@ -32,3 +32,32 @@ def test_template_response_requires_request(tmpdir):
     templates = Jinja2Templates(str(tmpdir))
     with pytest.raises(ValueError):
         templates.TemplateResponse("", {})
+
+
+def test_calls_context_processors(tmpdir, test_client_factory):
+    path = os.path.join(tmpdir, "index.html")
+    with open(path, "w") as file:
+        file.write("<html>Hello {{ username }}</html>")
```

Comment:
```suggestion
def test_calls_context_processors(tmp_path, test_client_factory):
    path = tmp_path / "index.html"
    path.write_text("<html>Hello {{ username }}</html>")
```

https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmp-path-fixture